### PR TITLE
docs: add documentation on configuration schema

### LINKED ISSUE #7455: Docs: Configuration schema in package.json
### Is your enhancement related to a problem? Please describe

See: https://github.com/containers/podman-desktop/issues/5561#issuecomment-2146142828

Sub-issue for adding configuration schema in package.json

### Describe the solution you'd like

Add configuration schema to docs

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
docs: add documentation on configuration schema

### What does this PR do?

Adds documentation on the configuration schema

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/7455

### How to test this PR?

N/A, docs.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
I am probably missing a few things in the "schema" example  if anyone can please verify for me

File: website/docs/extensions/developing/config.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+Within the schema, you may add any type of value such as `"foo":"bar"` which can be retrieved similar to the above TypeScript example.
+
+```json
+{
+  "contributes": {
+    "configuration": {
+      "title": "string",
+      "properties": {
+        "string": {
+          "string": "string",
```
Comment (lstocchi) [inline website/docs/extensions/developing/config.md]: to be removed?

File: website/docs/extensions/developing/config.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          "default": "string",
```
Comment (lstocchi) [inline website/docs/extensions/developing/config.md]: The default value really depends on the type value.
If type: "number" the default is a number (default: 10), if type: boolean the default is true or false. Not sure how to write it but string seems wrong to me

File: website/docs/extensions/developing/config.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+          "type": "string",
+          "format": "string",
+          "minimum": "string or int",
+          "maximum": "string or int",
+          "description": "string",
+          "scope": "string"
```
Comment (lstocchi) [inline website/docs/extensions/developing/config.md]: scope can also be an array like 
```
"scope": [
            "DEFAULT",
            "Onboarding"
          ],
```

File: website/docs/extensions/developing/config.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        }
```
Comment (lstocchi) [inline website/docs/extensions/developing/config.md:86]: It is missing 
hidden: boolean
placeholder: string
markdownDescription: string
readonly: boolean
enum: array
step: number
when: string

**Comment by cdrage** [comment]
Thank @lstocchi I really was missing a lot, thank you for the review! I've updated it with the changes.

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**